lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1704,8 +1703,13 @@ export class UserAgentApplication {
       this.cacheStorage.setItem(JSON.stringify(accessTokenKey), JSON.stringify(accessTokenValue));
 
       accessTokenResponse.accessToken  = parameters[Constants.accessToken];
-      accessTokenResponse.expiresIn = expiresIn;
       accessTokenResponse.scopes = consentedScopes;
+      let exp = Number(expiresIn);
+      if (exp) {
+        accessTokenResponse.expiresIn = new Date(Date.now() + (exp * 1000));
```

Comment:
Is there a reason why this isn't consistent here and in the else case?
Should this be new Date(Date.now() + (exp * 1000)); or new Date(exp * 1000);?

Reply:
In the else case we are getting the exp param from the token. That is the Numeric Date (in relation to 1-1-1970). In the if case we get expiresIn, a param from AAD which gives the lifetime of the token in seconds (i.e. 3600). The else case is more accurate, but we can only if it is an idToken since we do not unpack accessTokens.

Reply:
why not just use a ms time ?

Reply:
Want to be consistent with other MSALs here, where we are returning Date format.